tests/test_kernel32/test_handle.py

```diff
@@ -120,3 +124,162 @@ def test_process_dies_before_timeout(self):
+        expected = self._expected_inheritance()
+        self.assertEqual(inherit, expected)
+
+
+class TestSetHandleInformation(TestCase):
+    """
+    Tests for :func:`pywincffi.kernel32.SetHandleInformation`
+    """
+    def _set_handle_info_file(self, inherit, check=False):
+        _, library = dist.load()
+        tempdir = tempfile.mkdtemp()
+        self.addCleanup(os.rmdir, tempdir)
+        filename = os.path.join(tempdir, "test_file")
+        with open(filename, "w") as test_file:
+            self.addCleanup(os.unlink, filename)
+            test_file.write("data")
+            file_handle = handle_from_file(test_file)
+            SetHandleInformation(
+                file_handle,
+                library.HANDLE_FLAG_INHERIT,
+                inherit
+            )
+            if check:
+                result = GetHandleInformation(file_handle)
+                self.assertEqual(inherit, result)
+
+    def test_set_handle_info_file_noinherit(self):
+        self._set_handle_info_file(0)
+
+    def test_set_handle_info_file_inherit(self):
+        self._set_handle_info_file(1)
+
+    def test_set_get_handle_info_file_noinherit(self):
+        self._set_handle_info_file(0, check=True)
+
+    def test_set_get_handle_info_file_inherit(self):
+        self._set_handle_info_file(1, check=True)
+
+    def _set_handle_info_socket(self, inherit, check=False):
+        ffi, library = dist.load()
+        sock = socket.socket()
+        self.addCleanup(sock.close)
+        sock_handle = ffi.cast("void *", sock.fileno())
+        SetHandleInformation(
+            sock_handle,
+            library.HANDLE_FLAG_INHERIT,
+            inherit
+        )
+        if check:
+            result = GetHandleInformation(sock_handle)
+            self.assertEqual(inherit, result)
+
+    def test_set_handle_info_socket_noinherit(self):
+        self._set_handle_info_socket(0)
+
+    def test_set_handle_info_socket_inherit(self):
+        self._set_handle_info_socket(1)
+
+    def test_set_get_handle_info_socket_noinherit(self):
+        self._set_handle_info_socket(0, check=True)
+
+    def test_set_get_handle_info_socket_inherit(self):
+        self._set_handle_info_socket(1, check=True)
+
+
+class TestSetHandleInformationForks(TestCase):
+    """
+    Integration tests for :func:`pywincffi.kernel32.SetHandleInformation`
+    """
+    def test_file_rename_after_fork(self):
+        _, library = dist.load()
+        tempdir = tempfile.mkdtemp()
+        self.addCleanup(shutil.rmtree, tempdir, ignore_errors=True)
+        filename = os.path.join(tempdir, "original_name")
+        with open(filename, "w") as test_file:
+            test_file.write("data")
+            file_handle = handle_from_file(test_file)
+            # prevent file_handle inheritance
+            SetHandleInformation(file_handle, library.HANDLE_FLAG_INHERIT, 0)
+            # spawn child while test_file is open
+            p = subprocess.Popen(
```

Comment:
I prefer the approach I took, which is totally synchronised: launching a Python sub-process controlling its IO and, later, closing its STDIN and waiting for its exit.

The `time.sleep` approach becomes asynchronous and nothing in the test enforces the child process to be running while the tested operations are executed.